### PR TITLE
Fix endpoint URL and Output wrapped

### PR DESCRIPTION
**Pull request by jderusse**
This PR fixes 2 things:

* [x] Endpoint URL. the API.Json files contains a "http.requestUri" field that define the pattern of the endpoint (ie. `/{Bucket}/{Key+}`
* [x] Wrapped output. Example SQS.GetQueueUrl

### CODE REVIEW

File: build/Generator/ApiGenerator.php [hunk 1]
```diff
@@ -81,8 +81,8 @@ public function generateResultClass(array $shapes, $service, $baseNamespace, $cl
             // Add trait only if file does not exists
             $traitFilename = \sprintf('%s/%s/Result/%s.php', $this->srcDirectory, $service, $traitName);
             if (!\file_exists($traitFilename)) {
-                $this->createOutputTrait($baseNamespace, $traitName, $members, $traitFilename);
             }
+                $this->createOutputTrait($baseNamespace, $traitName, $members, $wrapper, $traitFilename);
```
Comment (Nyholm) [inline build/Generator/ApiGenerator.php]: This is debug, right?

File: src/Core/AbstractApi.php [hunk 2]
```diff
@@ -109,15 +109,17 @@ private function fillEndpoint(?string $endpoint): string
     /**
      * Fallback function for getting the endpoint. This could be overridden by any APIClient.
      *
-     * @param array $uri   parameters that should go in the URI
+     * @param string $uri   parameters that should go in the URI
      * @param array $query parameters that should go in the query string
      */
-    protected function getEndpoint(array $uri, array $query):?string
+    protected function getEndpoint(string $uri, array $query):?string
```
Comment (Nyholm) [inline src/Core/AbstractApi.php]: There is difference between `getEndpoint` and `fillEndpoint`. 

I wanted `getEndpoint` to be extended by special apis like s3. They have a custom URL..
But Im not super happy with my solution.

Comment (jderusse) [reply src/Core/AbstractApi.php]: I've the feeling that endpoints don't have to be customized (everything is available in API.json file).
I've tested my PR (which removes getEndpoint from Client) anbd both S3 and SQS works.

File: build/Generator/ApiGenerator.php (hunk 1 above)
Comment (jderusse) [reply build/Generator/ApiGenerator.php]: yep, sorry

**Pull request merged**